lib/req-res/responder.js

```diff
@@ -36,44 +36,63 @@ util.inherits( Responder, Rabbus.Responder );
 
 // Instance Methods
 // ----------------
-Responder.prototype.inbound = function ( message, respond, callback ) {
-	var statusOptions = {
-		'log'         : this.logger,
-		'messageType' : this.messageType,
-		'respond'     : respond,
-		'message'     : message,
-		'emitter'     : this
-	};
-
-	if ( !this.vSchema || this.vSchema === 'undefined' ) {
-		// if no validation, invoke callback immediately
-		callback( message, requestStatus( statusOptions ) );
-	} else {
+Responder.prototype.inbound = function ( message, send, statusOptions, callback ) {
+	return new Promise( function ( resolve, reject ) {
+		this.on( 'error', function ( consumeError ) {
+			reject( consumeError );
+		} );
 
-		// perform validation if there is a validation schema
-		validate( {
-			'value'   : message,
-			'schema'  : this.vSchema,
-			'options' : this.vOptions
-		} )
-		.then( function ( data ) {
-			callback( data, requestStatus( statusOptions ) );
-		} )
-		.catch( function ( error ) {
-			// send fail
-			respond( {
-				'status' : 'fail',
-				'data'   : error.message
+		if ( !this.vSchema || this.vSchema === 'undefined' ) {
+			// if no validation, invoke callback immediately
+			callback( message, send );
+		} else {
+			// perform validation if there is a validation schema
+			validate( {
+				'value'   : message,
+				'schema'  : this.vSchema,
+				'options' : this.vOptions
+			} )
+			.then( function ( data ) {
+				callback( data, send );
+			} )
+			.catch( function ( error ) {
+				// check if it's validation error
+				if ( error.validate ) {
+					send.fail( error.message );
+					return clearTimeout( statusOptions.timeout );
+				}
+				reject( error );
```

Comment:
it will because it is being catch here https://github.com/yamii/lapin/blob/master/lib/req-res/responder.js#L93

that catch is also use for this one https://github.com/yamii/lapin/blob/master/lib/req-res/responder.js#L42